pkg/nslookup/nslookup_test.go

```diff
@@ -0,0 +1,360 @@
+package nslookup
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/zmap/dns"
+	"github.com/zmap/zdns/pkg/miekg"
+	"github.com/zmap/zdns/pkg/zdns"
+)
+
+var nsRecords = make(map[string]miekg.Result)
+
+var errCode = zdns.STATUS_NOERROR
+
+// Mock the actual Miekg lookup.
+func (s *Lookup) DoMiekgLookup(question miekg.Question, nameServer string) (interface{}, []interface{}, zdns.Status, error) {
+	if res, ok := nsRecords[question.Name]; ok {
+		return res, nil, errCode, nil
+	} else {
+		return nil, nil, zdns.STATUS_NXDOMAIN, nil
+	}
+}
+
+var mockResults = make(map[string]miekg.IpResult)
+var protocolStatus = zdns.STATUS_NOERROR
+
+func (s *Lookup) DoTargetedLookup(l LookupClient, name, nameServer string, lookupIpv4 bool, lookupIpv6 bool) (interface{}, []interface{}, zdns.Status, error) {
+	retv := miekg.IpResult{}
+	if res, ok := mockResults[name]; ok {
+		if lookupIpv4 {
+			retv.IPv4Addresses = res.IPv4Addresses
+		}
+		if lookupIpv6 {
+			retv.IPv6Addresses = res.IPv6Addresses
+		}
+		return retv, nil, protocolStatus, nil
+	} else {
+		return nil, nil, zdns.STATUS_NXDOMAIN, nil
+	}
+}
+
+type minimalServerRecords struct {
+	IPv4Addresses []string
+	IPv6Addresses []string
+}
+
+func InitTest(t *testing.T) (*zdns.GlobalConf, *GlobalLookupFactory, *RoutineLookupFactory, zdns.Lookup) {
```

Comment:
Here as well, we'll likely want to test some failure modes. Additionally, we may be able to abstract some of that away to the util package if the test logic is shared across modules

Reply:
Added failure modes. I don't want to move `InitTest` to util since `GlobalLookupFactory` and `RoutineLookupFactory` differs across modules.

Reply:
Concur, I must've overlooked the differences.